urla958522fc1_spirosikmd_inmdsy_08_tgz-pJ8-TesterAppJ8/src/ft2client.java

```diff
@@ -0,0 +1,46 @@
+import java.io.*;
+import java.net.*;
+import java.util.*;
+public class ft2client {
+
+	@SuppressWarnings("deprecation")
+	public static void main(String[] args) throws IOException {
+		{
+			 Socket s=null;
+             BufferedReader get=null;
+             PrintWriter put=null;
+             try
+             { 
+                 s=new Socket("127.0.0.1",8081);
+                 get=new BufferedReader(new InputStreamReader(s.getInputStream()));
+                 put=new PrintWriter(s.getOutputStream(),true);
+             }  
+             catch(Exception e)
+             {
+                 System.exit(0);
+             }
+             String u,f;
+             System.out.println("Enter the file name to transfer from server:");
+             DataInputStream dis=new DataInputStream(System.in);
+             f=dis.readLine();
+             put.println(f);
+             File f1=new File(f);
+             FileOutputStream  fs=new FileOutputStream(f1);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fs*. The resource is referenced at line 32. The resource is closed at line 34. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure *fs* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).